libnavigation-core/src/main/java/com/mapbox/navigation/core/routeoptions/RouteOptionsUpdater.kt

```diff
@@ -46,94 +47,110 @@ class RouteOptionsUpdater {
-                                return@radiusesList emptyList<Double>()
-                            }
-                            mutableListOf<Double>().also {
-                                it.addAll(radiusesList.subList(index, coordinatesList.size))
-                            }
+            val location = locationMatcherResult.enhancedLocation
+            optionsBuilder
+                .coordinatesList(
+                    coordinatesList
+                        .subList(nextCoordinateIndex, coordinatesList.size)
+                        .toMutableList()
+                        .apply {
+                            add(0, Point.fromLngLat(location.longitude, location.latitude))
                         }
+                )
+                .bearingsList(
+                    getUpdatedBearingList(
+                        remainingCoordinates,
+                        nextCoordinateIndex,
+                        location.bearing.toDouble(),
+                        routeOptions.bearingsList(),
                     )
-                    .approachesList(
-                        let approachesList@{
-                            val approachesList = routeOptions.approachesList()
-                            if (approachesList.isNullOrEmpty()) {
-                                return@approachesList emptyList<String>()
-                            }
-                            mutableListOf<String>().also {
-                                it.addAll(approachesList.subList(index, coordinatesList.size))
-                            }
+                )
+                .radiusesList(
+                    let radiusesList@{
+                        val radiusesList = routeOptions.radiusesList()
+                        if (radiusesList.isNullOrEmpty()) {
+                            return@radiusesList emptyList<Double?>()
                         }
-                    )
-                    .apply {
-                        if (routeOptions.profile() == DirectionsCriteria.PROFILE_DRIVING_TRAFFIC) {
-                            snappingIncludeClosuresList(
-                                routeOptions.snappingIncludeClosuresList()
-                                    .withFirstTrue(remainingWaypoints, index, coordinatesList.size)
+                        mutableListOf<Double?>().also {
+                            it.addAll(
+                                radiusesList.subList(
+                                    nextCoordinateIndex - 1,
+                                    coordinatesList.size
+                                )
                             )
-                            snappingIncludeStaticClosuresList(
-                                routeOptions.snappingIncludeStaticClosuresList()
-                                    .withFirstTrue(remainingWaypoints, index, coordinatesList.size)
+                        }
+                    }
+                )
+                .approachesList(
```

Comment:
Approaches should be specified per waypoint, not per coordinate (capturing from https://docs.mapbox.com/api/navigation/directions/#optional-parameters), so it should use `getUpdatedWaypointsList`.

Reply:
We can address separately, let's cut a ticket.

Reply:
`NAVAND-834`